pkg/apis/provisioning/v1alpha3/provisioner.go

```diff
@@ -26,15 +29,11 @@ import (
 // provisioner specification (defaults) and pod scheduling constraints
 // (overrides). A single provisioner is capable of managing highly diverse
 // capacity within a single cluster and in most cases, only one should be
-// necessary. For advanced use cases like workload separation and sharding, it's
-// possible to define multiple provisioners. These provisioners may have
-// different defaults and can be specifically targeted by pods using
+// necessary. It's possible to define multiple provisioners. These provisioners
+// may have different defaults and can be specifically targeted by pods using
 // pod.spec.nodeSelector["karpenter.sh/provisioner-name"]=$PROVISIONER_NAME.
 type ProvisionerSpec struct {
-	// Cluster that launched nodes connect to.
-	Cluster Cluster `json:"cluster"`
```

Comment:
`Cluster` is not provider specific and common to all providers, why is it removed here?

Maybe we can consider making it an optional field, because usually all the information provided in `Cluster` can also be discovered from the in-cluster configuration. 
- If omitted, use in-cluster configuration to connect to the cluster karpenter is running in.
- If present, use provided information to connect to the cluster, which might be a remote cluster.

```
Cluster *Cluster json:"cluster,omitempty"
```

Reply:
As discussed in some other PRs, we've decided to make the mechanism for connecting to a cluster cloud provider specific. Other providers don't necessarily have a concept of cluster name and in some cases, cloud providers may be able to connect with just a dns endpoint or nothing at all. It's also possible that providers could implement a provider specific discovery mechanism to determine all of this data. For example, you could imagine CAPI using a CRD Ref to a Cluster CRD.

I'll have a doc detailing this soon.